src/components/app/list-new/AppList.tsx

```diff
@@ -321,28 +321,109 @@ export default function AppList() {
         history.push(url);
     }
 
-    const applyFilter = (type: string, list: FilterOption[], selectedAppTab : string = undefined): void => {
+    /**
+     * This function will return filters to be applied for filter types - CLUSTER & NAMESPACE (query param - namespace)
+     *
+     * @param ids - currently selected/checked items from filters list
+     * @param filterType - type of filter
+     * @param query - current query params
+     * @returns string - filters to be applied
+     */
+    const getUpdatedFiltersOnNamespaceChange = (ids: (string | number)[], filterType: string, query: Record<string, string>): string => {
+        /**
+         * Step 1: Return currently selected/checked items from filters list as string if
+         * - There are no query params
+         * - There is no namespace query param i.e. this is the first time selecting the cluster filter
+         */
+        if (
+            !query ||
+            Object.keys(query).length <= 0 ||
+            typeof query[AppListConstants.FilterType.NAMESPACE] === 'undefined'
```

Comment:
We should handle empty as well. 
!query[AppListConstants.FilterType.NAMESPACE]

(As after removing the filters, namespace in query becomes empty)

Reply:
Done